qip-0009.md

```diff
@@ -0,0 +1,131 @@
+2. $Intrinsic Difficulty (id)$ - $ \text{length of hash field}  - \log_2(\text{PoW Hash of the block})$ 
+
+### Rank ($\mu$) of a Block
+Proof-of-Work function of a Valid Block in the blockchain satisfies the Mining Target(T) i.e $id \leq T$ . Traditional Interlinks are organized into different levels. If $id \leq
+\frac{T}{2^{k \times \mu}}$ we say the block is of rank $\mu$. k is a compression
+parameter and determines the range between interlinks. This is a small
+deviation from the equation in [1] $id \leq \frac{T}{2^{\mu}}$. Rank in Quai is analogous to Level in paper[1]. Since Quai has three layers namely Prime, Region, Zone. Prime chain acts as the base chain and has rank of 0. Region chain has a rank of -1 and Zone chain has a rank of -2 by definition.
+
+Genesis Block is set to have $id=0$ and $\mu=\inf$.
+
+## Specification
+
+### Interlink Data structure
+
+Interlinks is an array of hashes of the interlink blocks for each rank between 1 and max length of $L$.
+
+$\text{InterLinks}[\mu-1]$ with $0 <= len(Interlinks) <= L $
+
+This QIP proposes to add new field to the header:
+* Patricia Merkle root of the interlinks array. This will be recomputed and verified on each prime block.
+
+```{go}
+Header {
+  ...
+  interlinkRootHash Hash
+  ...
+}
+```
+
+## Prover and Verifier Design
+
+Benefit of using non interactive proofs is that a verifier that doesn't have the
+state of the blockchain can confirm with a very high probability that a
+transaction was included in a particular Block. Since this process is non
+interactive there are two independent parties that participate in this process.
+**Prover** and **Verifier**. Prover produces the proofs, and Verifier verifies
+the proof. 
+
+Paper [1] goes over how this can done in a single chain setting. In [1] the proof is separated into two parts, suffix proofs and infix proofs. 
+1. Suffix proofs show the link to the highest rank of interlink close to the Block that also satisfies the $m$ (security parameter). $m$ refers to the minimum number of Interlink blocks at a particular rank needed to generate the suffix proof. If we don't have atleast $m$ blocks from a particular rank, we default to $\mu=-2$, i.e generating the proofs with just the zone blocks.
+2. Infix proofs shows the link between the Block for which the proof is generated to the first block that satisfies the $m$ criteria.
+
+### InterlinkProof structure
+
+InterlinkProof includes
+1. Continuous chain of headers that attach the Block to the chain Tip (i.e suffix proof + infix proof) and also satisfies the $m$ security parameter 
+2. MerkleProof of inclusion of txHash in the Block, and txHash
+
+```{go}
+InterlinkProof {
```

Comment:
What is this interlink proof structure? Is this supposed to be a NiPoPoW?

It also needs the interlink structures of each header

TxHash and merkle proof are not necessarily the data to be proved, and even if they are, they would be independent of the NiPoPoW.

The interlink exists to construct a NiPoPoW which proves some _block_ has some _work_ built on top. Proving other things about that block (tx inclusion, account state at block, etc) are independent of this QIP.